### ISSUE TITLE
Incorrect error "Time difference between steps has been idenfied as larger than 1 - set allow_missing_timesteps=True"

### ISSUE THREAD

**Issue by cowwoc** [jdb78/pytorch-forecasting#592]
- PyTorch-Forecasting version: 0.9.0

I get "Time difference between steps has been idenfied as larger than 1 - set allow_missing_timesteps=True" when in fact `df_index["time_diff_to_next"] == 0`.

Either the error message should be improved or this case should be accepted.

**Comment by jdb78** [comment]
So you have the same time step twice? The diff to the next time step should be 1.

**Comment by cowwoc** [comment]
@jdb78 Yes. I had code that mistakenly collapsed two different times into the same time step because they had different minutes and the code expected a minimum of 1 month between steps.

It sounds the error message just needs to be improved in this case.

**Comment by antolu** [comment]
A bit necro here, but I run into the same error now, but with what I believe should be a correct time range.

Sample code:

```
DATASET = 'SPS-BTRAIN-20230315-091654---20230315-094954-1kHz.parquet'
DOWNSAMPLE = 50

df = pd.read_parquet(DATASET)

df = df.iloc[::DOWNSAMPLE]
df['time_idx'] = np.arange(len(df))

dataset = TimeSeriesDataSet(
    df,
    time_idx='time_idx',
    target=FIELD,
    group_ids=[CURRENT],
    max_encoder_length=1000,
    max_prediction_length=1000,
    time_varying_known_reals=[CURRENT],
    time_varying_unknown_reals=[FIELD],
    allow_missing_timesteps=False,
)
```
results in `AssertionError: Time difference between steps has been idenfied as larger than 1 - set allow_missing_timesteps=True` even though I explicitly just generated a new time axis....

This is on v0.10.3.

**Comment by techosystem** [comment]
@jdb78 Please I got the same error "AssertionError: Time difference between steps has been identified as larger than 1 - set allow_missing_timesteps=True" while trying to reproduce Pytorch forecasting code with different data.
I set the add_nan = True but I got an error validation_dataloader  while training the model, RuntimeError: index 1122 is out of bounds for dimension 0 with size 25. I have checked my time index and found out they are not equal. Is there a way I can get out of this error?
